pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -391,7 +400,8 @@ func (lb *ListenerBuilder) buildVirtualInboundListener(configgen *ConfigGenerato
 		isTransparentProxy = proto.BoolTrue
 	}
 
-	actualWildcard, _ := getActualWildcardAndLocalHost(lb.node)
+	podIP := lb.node.IPAddresses[0]
```

Comment:
A bit concerned about this. Workload may have many IPs. Can this break anything? @lambdai

Reply:
One thing that is also unfortunate is it completely kills our ability to cache. We aren't caching this today but in the future we could

Reply:
For workloads that have many IPs in the same family (e.g. with Multus), the iptables DNAT rules will ensure traffic on any of those IPs is forwarded to the IP on which Envoy is listening.

I'm more concerned about dual-stack IPv4/IPv6 use cases. For these, we would need to duplicate the inbound listener entirely or wait for https://github.com/envoyproxy/envoy/pull/19367 as you had mentioned. Because of this, I'm considering descoping dual-stack from this change. Dual stack use cases would continue listening on wildcard for now. This would mean outbound on 15006 would remain broken for dual-stack use cases, but we would avoid mass listener duplication. Thoughts?

Reply:
@howardjohn, I realize now that dual stack isn't yet supported by Istio, so this shouldn't be a concern for now, right?

Reply:
We are working to add it, and this will be a concern. 

I would also be concerned about the change from REDIRECT to DNAT - we also use this for VMs, where AFAIK the IP address may change (on Pods it's almost always fixed - at least the first one). IPv6 addresses do change dynamically quite often ( network prefix ). 

Anything that involves using the IP address is dangerous - in particular since we don't regenerate the bootstrap and can't send updates to istiod.

Reply:
I see. If we can't rely on a fixed IP address, then this solution would require istiod to watch for changes in IP address and update the envoy config accordingly. Doesn't seem ideal.

The only reason this is needed is to prevent outbound on port 15006 from being redirected to the inbound listener. Perhaps a better solution would be for Envoy to support a "don't redirect to me" flag on listeners? If such a thing existed, we could continue to listen on wildcard address for the inbound listener without disrupting outbound traffic.